src/SFA.DAS.EmployerAccounts.UnitTests/Queries/GetSingleDraftApprenticeship/WhenIGetSingleDraftApprenticeship.cs

```diff
@@ -0,0 +1,78 @@
+using SFA.DAS.Encoding;
+using SFA.DAS.Validation;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using SFA.DAS.NLog.Logger;
+using SFA.DAS.CommitmentsV2.Types.Dtos;
+
+namespace SFA.DAS.EmployerAccounts.UnitTests.Queries.GetSingleDraftApprenticeship
+{
+    public class WhenIGetSingleDraftApprenticeship : QueryBaseTest<GetSingleDraftApprenticeshipRequestHandler, GetSingleDraftApprenticeshipRequest, EmployerAccounts.Queries.GetSingleDraftApprenticeship.GetSingleDraftApprenticeshipResponse>
+    {
+        public override GetSingleDraftApprenticeshipRequest Query { get; set; }
+        public override GetSingleDraftApprenticeshipRequestHandler RequestHandler { get; set; }
+        public override Mock<IValidator<GetSingleDraftApprenticeshipRequest>> RequestValidator { get; set; }
+
+        private Mock<ICommitmentV2Service> _commitmentV2Service;
+        private Mock<IEncodingService> _encodingService;
+        private Mock<ILog> _logger;
+        private long _cohortId;
+
+        [SetUp]
+        public void Arrange()
+        {
+            SetUp();
+
+            _cohortId = 123;
+            _logger = new Mock<ILog>();
+
+            _commitmentV2Service = new Mock<ICommitmentV2Service>();
+            _commitmentV2Service.Setup(m => m.GetDraftApprenticeships(_cohortId))
+                .ReturnsAsync(new SFA.DAS.CommitmentsV2.Api.Types.Responses.GetDraftApprenticeshipsResponse() 
+                { DraftApprenticeships = new List<DraftApprenticeshipDto>() { new DraftApprenticeshipDto { Id = 4} } });
+            _encodingService = new Mock<IEncodingService>();
+            _encodingService.Setup(x => x.Encode(It.IsAny<long>(), EncodingType.ApprenticeshipId)).Returns((long y, EncodingType z) => y + "_Encoded");
+
+            RequestHandler = new GetSingleDraftApprenticeshipRequestHandler(RequestValidator.Object, _logger.Object, _commitmentV2Service.Object, _encodingService.Object);
+
+            Query = new GetSingleDraftApprenticeshipRequest
+            {
+                CohortId = _cohortId
+            };
+        }
+      
+        [Test]
+        public override async Task ThenIfTheMessageIsValidTheValueIsReturnedInTheResponse()
+        {
+            //Act
+            var response = await RequestHandler.Handle(Query);
+
+            //Assert            
+            Assert.IsNotNull(response.DraftApprenticeshipsResponse);
+            Assert.AreEqual("4_Encoded", response.HashedDraftApprenticeshipId);
+        }
+
+
+        [Test]
+        public async Task ThenIfTheMessageIsValidTheServiceIsCalled()
+        {
+            //Act
+            await RequestHandler.Handle(Query);
+
+            //Assert
+            _commitmentV2Service.Verify(x => x.GetDraftApprenticeships(_cohortId), Times.Once);
+        }
+
+        public override Task ThenIfTheMessageIsValidTheRepositoryIsCalled()
+        {
+            throw new NotImplementedException();
```

Comment:
Should this be implemented?

Reply:
Base class got this definition, hence inheriting and implementation is not required in this case ,  not doing anything to do with repository.